### DISCUSSION TITLE
FAISS vs milvus memory consumption? Which on is more efficient?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#22892] [Q&A and General discussion]
FAISS vs milvus memory consumption? Which on is more efficient?

**Comment by xiaofan-luan** [comment]
FAISS is one of the index type supported by milvus, if you use FAISS then memory consumption is the same.

Milvus takes more memory because it is system rather than library so there is some overhead,

**Reply by martinenkoEduard** [reply]
When using flat index does milvus use FAISS under the hood?

**Reply by xiaofan-luan** [reply]
Yes, all the IVF series is from FAISS, Milvus also support Annoy, HNSW and other index types

**Reply by martinenkoEduard** [reply]
So if there are several worker nodes, the data will be distributed across several faiss instances and will be aggregated in one place when the search is performed? Is it the right guess?

**Reply by xiaofan-luan** [reply]
milvus split data into segments.
each segments will retrieve topk and final result is reduced at proxy
